libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -153,4 +154,35 @@ internal class MapboxDirectionsSession(
     override fun shutdown() {
         router.shutdown()
     }
+
+    /**
+     * Temporary method for handle routes with multiple waypoints.
+     * Current Directions API doesn't support more that one waypoint.
```

Comment:
The Directions API should support at least one intermediate waypoint (at least three total) [depending on the profile](https://docs.mapbox.com/api/navigation/#directions-api-restrictions-and-limits).